src/evaluate/evaluator/base.py

```diff
@@ -278,14 +284,49 @@ def check_required_columns(self, data: Union[str, Dataset], columns_names: Dict[
                     f"Invalid `{input_name}` {column_name} specified. The dataset contains the following columns: {data.column_names}."
                 )
 
-    def prepare_data(self, data: Union[str, Dataset], input_column: str, label_column: str):
+    @staticmethod
+    def get_dataset_split(data, split):
+        """
+        Infers which split to use if None is given.
+
+        Args:
+             data (`str`): Name of dataset
+             split (`str`, defaults to None): Split to use
+        Returns:
+            `split`: `str` containing which split to use
+        """
+        if split is None:
+            split = choose_split(data)
+            logger.warning(f"Dataset split not defined! Automatically evaluating with split: {split.upper()}")
+        return split
+
+    def load_data(self, data: Union[str, Dataset], split: str = None):
+        """
+        Load dataset with given split.
+        Args:
+            data (`Dataset` or `str`, defaults to None): Specifies the dataset we will run evaluation on. If it is of
+            type `str`, we treat it as the dataset name, and load it. Otherwise we assume it represents a pre-loaded dataset.
+            split (`str`, defaults to None):
+                User-defined dataset split by name (e.g. train, validation, test). Supports slice-split (test[:n]).
+                If not defined and data is a `str` type, will automatically select the best one via `choose_split()`.
+        Returns:
```

Comment:
Looking at the doc-builder code it seems to happen in the block that parses returns - I think this empty return could be the culprit.

Reply:
Thanks for the catch! That was it.